x/emissions/keeper/msgserver/msg_server_params.go

```diff
@@ -119,20 +110,14 @@ func (ms msgServer) UpdateParams(ctx context.Context, msg *types.MsgUpdateParams
 	if len(newParams.MaxRetriesToFulfilNoncesReputer) == 1 {
 		existingParams.MaxRetriesToFulfilNoncesReputer = newParams.MaxRetriesToFulfilNoncesReputer[0]
 	}
-	if len(newParams.TopicPageLimit) == 1 {
-		existingParams.TopicPageLimit = newParams.TopicPageLimit[0]
-	}
-	if len(newParams.MaxTopicPages) == 1 {
-		existingParams.MaxTopicPages = newParams.MaxTopicPages[0]
-	}
 	if len(newParams.RegistrationFee) == 1 {
 		existingParams.RegistrationFee = newParams.RegistrationFee[0]
 	}
-	if len(newParams.DefaultLimit) == 1 {
-		existingParams.DefaultLimit = newParams.DefaultLimit[0]
+	if len(newParams.DefaultPageLimit) == 1 {
+		existingParams.DefaultPageLimit = newParams.DefaultPageLimit[0]
 	}
-	if len(newParams.MaxLimit) == 1 {
-		existingParams.MaxLimit = newParams.MaxLimit[0]
+	if len(newParams.MaxPageLimit) == 1 {
+		existingParams.MaxPageLimit = newParams.MaxPageLimit[0]
 	}
 	if len(newParams.MinEpochLengthRecordLimit) == 1 {
 		existingParams.MinEpochLengthRecordLimit = newParams.MinEpochLengthRecordLimit[0]
```

Comment:
Just below here, just before `SetParams`, is the most crucial place for validation of these params to go...

Reply:
fuck

good catch lol